pkg/scheduler/api/node_info.go

```diff
@@ -229,8 +229,7 @@ func (ni *NodeInfo) AddTask(task *TaskInfo) error {
 
 	key := PodKey(task.Pod)
 	if _, found := ni.Tasks[key]; found {
-		return fmt.Errorf("task <%v/%v> already on node <%v>",
-			task.Namespace, task.Name, ni.Name)
+		return nil
```

Comment:
why return nil?

Reply:
I am not sure the logic before, it looks this happens when calling AddTask multi-times. This should be a warning, and we should fix the call site.

This pr is just trying to verify if this way is viable

Reply:
> This should be a warning, and we should fix the call site.

If we return nil, how to know the issue there?